src/components/SourcesTree.js

```diff
@@ -83,6 +92,12 @@ let SourcesTree = React.createClass({
     if (!nodeHasChildren(item)) {
       this.props.selectSource(item.contents.get("id"));
     }
+
+    // If listItems is not empty, focusedItem would be set incorrectly in
+    // ManagedTree.js's expandListItems function
+    if (this.state.listItems && this.state.listItems.length > 0) {
+      this.setState({ listItems: [] });
+    }
```

Comment:
hmm, @arthur801031 this *also* seems strange.

I think it's here because `componentWillReceiveProps` does not check to see if `listItems` has changed. 
* perhaps check to see if listItems is different
* also could, `componentWillReceiveProps` be `componentDidUpdate` perhaps not, just curious.